### ISSUE TITLE
In 1.64 split-debuginfo = "unpacked" was ignored on Windows while in 1.65 it fails to build

### ISSUE THREAD

**Issue by Yatekii** [rust-lang/rust#103976]
### Code

I tried this code:

```toml
[profile.dev]
split-debuginfo = "unpacked"
```

I expected to see this happen: The compilation on windows succeds but does not actually use the feature.

Instead, this happened: The compilation on windows lets me know this feature is unavailable.

### Version it worked on

It most recently worked on: Rust 1.64

### Version with regression

`rustc --version --verbose`:
```
rustc 1.65.0 (897e37553 2022-11-02)
```

This is the `Cargo.toml` in question: https://github.com/probe-rs/probe-rs/blob/master/Cargo.toml#L21.
This is the CI pipeline that started to fail with th 1.65 release: https://github.com/probe-rs/probe-rs/actions/runs/3390410063/jobs/5634519715

I am not sure if this would be a cargo or rustc issue. Pardon my ignorance.

Best,
Noah

**Comment by kpreid** [comment]
I hit this in my CI too.

Here are some other (not Windows-specific) reports of new failures when using `split-debuginfo`: #100657, #103932

**Comment by apiraino** [comment]
@Jules-Bertholet I think #104104 was meant to close this. But this issue still reproduces, right?

**Comment by ehuss** [comment]
I'm going to close as I believe this is fixed in 1.68 via https://github.com/rust-lang/cargo/pull/11347. Feel free to comment if that doesn't seem to be the case.
